Extensions/CodeGeneration/CodeGenerationDisplayDriver.cs

```diff
@@ -146,7 +146,7 @@ private void AddSettingsWithout<T>(StringBuilder codeBuilder, JObject settings,
             var indentation = string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth));
 
             var filteredSettings = ((IEnumerable<KeyValuePair<string, JToken>>)settings)
-                .Where(setting => setting.Key != typeof(T).Name);
+                .Where(setting => !setting.Key.EqualsOrdinal(typeof(T).Name));

```

Comment:
All these equals instead of operators really give me C++ vibes and not in a good way. What is the justification here?

Reply:
I was going to ask what kind of comparison is being applied when using the equality operator, but [the docs](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/equality-operators#string-equality) seem to be clear:
> That is a case-sensitive ordinal comparison.

So in this case, the extension method would actually not be necessary.

Disclaimer: I really hope you don't mind me commenting here, but I'm really fond of learning and there are so many opportunities when I read your code comments, that I cannot let them pass.

Reply:
Thanks for the research, Oliver.

If it's only an opinion thing with no technical reason then I'm against Meziantou's opinion here and in favor of keeping the operator to "improve the readability of the code". ;) I've added a poll [here](https://teams.microsoft.com/l/message/19:525a1a3656c845ae92c5e2c0d3617cc7@thread.skype/1640684101116?tenantId=959ede5a-38e9-44e6-adc8-85553beb8220&groupId=d6a5d775-6fb9-41dc-a7bb-481d4f884046&parentMessageId=1640684101116&teamName=Lombiq%20Developers&channelName=General&createdTime=1640684101116).

Reply:
Thanks for the tip, this is useless then. I've reversed such changes.